### PR TITLE
zig: update install fn to force LLVM static linking

### LINKED ISSUE #8860: homebrew-provided LLVM/Clang prevents zig from building C files
Note from @andrewrk: A workaround for this is to use the tarballs from [the download page](ziglang.org/download/).

---------

zig version: `0.8.0-dev.2591+4b69bd61e`
macOS version: `11.3.1`

To repro, run this:
```
echo "#include <assert.h>" > file.c
zig cc -lc hello.c 
```

Response:
```bash
jarred@c-73-252-132-29 ~/Code [1]> zig cc -lc hello.c
hello.c:1:10: fatal error: cannot open file '/usr/local/Cellar/zig/HEAD-4b69bd6_1/lib/zig/include/assert.h': No such file or directory
#include <assert.h>
         ^
```

Passing the include path does not fix it:
```bash
zig cc -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include /Users/jarred/Code/esdev/src/deps/picohttpparser.c 
/Users/jarred/Code/esdev/src/deps/picohttpparser.c:27:10: fatal error: cannot open file '/usr/local/Cellar/zig/HEAD-4b69bd6_1/lib/zig/include/assert.h': No such file or directory
#include <assert.h>
```

This is consistent, whether it's `zig translate-c`, `zig cc`, `@cInclude`, or `exe.addCSourceFile` etc.

Initially, I thought it was something specific to my environment -- but after wiping my hard-drive & reinstalling macOS, it's hard to see how it could be. 

Note that llvm 12 is installed and this occurs whether it's a copy of zig built from homebrew or manually running `cmake`

This seems to work on zig 0.7.1, so this seems to be something that broke recently:
```
jarred@c-73-252-132-29 ~/Code> zig cc -lc hello.c
Undefined symbol: C entry: _main
symbol(s) not found
error: FileNotFound
```

After adding a main function, `zig 0.7.1` can build it, but zig on `HEAD` cannot

### LINKED ISSUE #78025: llvm+zig HEAD: current installation of LLVM 12 breaks zig HEAD
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
❯ brew config
HOMEBREW_VERSION: 3.1.9-8-g6f1398a
ORIGIN: https://github.com/Homebrew/brew
HEAD: 6f1398a5338d851151041c81562da38b62cceff3
Last commit: 6 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 051251736a98e9c97996b053c5b7c301b24a8352
Core tap last commit: 73 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 16
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 16-core 64-bit kabylake
Clang: 12.0.5 build 1205
Git: 2.30.1 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.3.1-x86_64
CLT: 12.5.0.22.9
Xcode: N/A

❯ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Homebrew's "sbin" was not found in your PATH but you have installed
formulae that put executables in /usr/local/sbin.
Consider setting your PATH for example like so:
  echo 'export PATH="/usr/local/sbin:$PATH"' >> ~/.zshrc
```

### 

- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.

### What were you trying to do (and why)?

Current Homebrew's LLVM 12 installation breaks `zig` HEAD by what seems an issue with mixing shared with static libs. In particular, `zig`'s `CMake` config script (which was not changed in between LLVM releases in a way that would affect this, at least not as far as I am aware) configures the project to link dynamically against LLVM libs, i.e., `-lLLVM-12`, while `libclang*` is found and designated for static linking.

**Tl;dr**

```
> brew install zig --HEAD
```

will produce a broken `zig` compiler.

### What happened (include all command output)?

This has very interesting runtime issues (probably a result of a convoluted network of static initializers as described [here](https://reviews.llvm.org/D61804)) which manifest themselves in `zig` no longer being able to act as a C/C++ compiler:

```
❯ zig cc hello.c
hello.c:1:10: fatal error: cannot open file '/Users/jakubkonka/dev/zig/build/lib/zig/include/stdio.h': No such file or directory
#include <stdio.h>
         ^
1 error generated.
```

Here `zig` was installed with `brew install zig --HEAD`. This error was reported to us upstream by multiple users on multiple occasions and here's the tracking issue https://github.com/ziglang/zig/issues/8860. Interestingly enough, this problem was not present with `llvm@11` as `llvm@11` reported LLVM built with static linking unlike `llvm@12`:

```
❯ /usr/local/opt/llvm@11/bin/llvm-config --libs
-lLLVMXRay -lLLVMWindowsManifest -lLLVMTableGen -lLLVMSymbolize -lLLVMDebugInfoPDB -lLLVMOrcJIT -lLLVMOrcError -lLLVMJITLink -lLLVMObjectYAML -lLLVMMCA -lLLVMLTO -lLLVMObjCARCOpts -lLLVMExtensions -lPolly -lPollyISL -lLLVMPasses -lLLVMCoroutines -lLLVMLineEditor -lLLVMLibDriver -lLLVMInterpreter -lLLVMFuzzMutate -lLLVMMCJIT -lLLVMExecutionEngine -lLLVMRuntimeDyld -lLLVMDWARFLinker -lLLVMDlltoolDriver -lLLVMOption -lLLVMDebugInfoGSYM -lLLVMCoverage -lLLVMXCoreDisassembler -lLLVMXCoreCodeGen -lLLVMXCoreDesc -lLLVMXCoreInfo -lLLVMX86Disassembler -lLLVMX86AsmParser -lLLVMX86CodeGen -lLLVMX86Desc -lLLVMX86Info -lLLVMWebAssemblyDisassembler -lLLVMWebAssemblyCodeGen -lLLVMWebAssemblyDesc -lLLVMWebAssemblyAsmParser -lLLVMWebAssemblyInfo -lLLVMSystemZDisassembler -lLLVMSystemZCodeGen -lLLVMSystemZAsmParser -lLLVMSystemZDesc -lLLVMSystemZInfo -lLLVMSparcDisassembler -lLLVMSparcCodeGen -lLLVMSparcAsmParser -lLLVMSparcDesc -lLLVMSparcInfo -lLLVMRISCVDisassembler -lLLVMRISCVCodeGen -lLLVMRISCVAsmParser -lLLVMRISCVDesc -lLLVMRISCVUtils -lLLVMRISCVInfo -lLLVMPowerPCDisassembler -lLLVMPowerPCCodeGen -lLLVMPowerPCAsmParser -lLLVMPowerPCDesc -lLLVMPowerPCInfo -lLLVMNVPTXCodeGen -lLLVMNVPTXDesc -lLLVMNVPTXInfo -lLLVMMSP430Disassembler -lLLVMMSP430CodeGen -lLLVMMSP430AsmParser -lLLVMMSP430Desc -lLLVMMSP430Info -lLLVMMipsDisassembler -lLLVMMipsCodeGen -lLLVMMipsAsmParser -lLLVMMipsDesc -lLLVMMipsInfo -lLLVMLanaiDisassembler -lLLVMLanaiCodeGen -lLLVMLanaiAsmParser -lLLVMLanaiDesc -lLLVMLanaiInfo -lLLVMHexagonDisassembler -lLLVMHexagonCodeGen -lLLVMHexagonAsmParser -lLLVMHexagonDesc -lLLVMHexagonInfo -lLLVMBPFDisassembler -lLLVMBPFCodeGen -lLLVMBPFAsmParser -lLLVMBPFDesc -lLLVMBPFInfo -lLLVMAVRDisassembler -lLLVMAVRCodeGen -lLLVMAVRAsmParser -lLLVMAVRDesc -lLLVMAVRInfo -lLLVMARMDisassembler -lLLVMARMCodeGen -lLLVMARMAsmParser -lLLVMARMDesc -lLLVMARMUtils -lLLVMARMInfo -lLLVMAMDGPUDisassembler -lLLVMAMDGPUCodeGen -lLLVMMIRParser -lLLVMipo -lLLVMInstrumentation -lLLVMVectorize -lLLVMLinker -lLLVMIRReader -lLLVMAsmParser -lLLVMFrontendOpenMP -lLLVMAMDGPUAsmParser -lLLVMAMDGPUDesc -lLLVMAMDGPUUtils -lLLVMAMDGPUInfo -lLLVMAArch64Disassembler -lLLVMMCDisassembler -lLLVMAArch64CodeGen -lLLVMCFGuard -lLLVMGlobalISel -lLLVMSelectionDAG -lLLVMAsmPrinter -lLLVMDebugInfoDWARF -lLLVMCodeGen -lLLVMTarget -lLLVMScalarOpts -lLLVMInstCombine -lLLVMAggressiveInstCombine -lLLVMTransformUtils -lLLVMBitWriter -lLLVMAnalysis -lLLVMProfileData -lLLVMObject -lLLVMTextAPI -lLLVMBitReader -lLLVMCore -lLLVMRemarks -lLLVMBitstreamReader -lLLVMAArch64AsmParser -lLLVMMCParser -lLLVMAArch64Desc -lLLVMMC -lLLVMDebugInfoCodeView -lLLVMDebugInfoMSF -lLLVMBinaryFormat -lLLVMAArch64Utils -lLLVMAArch64Info -lLLVMSupport -lLLVMDemangle

❯ /usr/local/opt/llvm@12/bin/llvm-config --libs
-lLLVM-12
```

**Tl;dr**

Installing `zig` HEAD and running tests will fail:

```
> brew install zig --HEAD
> brew test zig
==> Testing zig
==> /usr/local/Cellar/zig/HEAD-2f538f3_1/bin/zig build-exe hello.zig
==> ./hello
==> /usr/local/Cellar/zig/HEAD-2f538f3_1/bin/zig cc hello.c -o hello
Last 15 lines from /Users/jakubkonka/Library/Logs/Homebrew/zig/test.02.zig:
2021-05-25 23:16:22 +0200

/usr/local/Cellar/zig/HEAD-2f538f3_1/bin/zig
cc
hello.c
-o
hello

hello.c:1:10: fatal error: cannot open file '/usr/local/include/stdio.h': No such file or directory
#include <stdio.h>
         ^
1 error generated.
Error: zig: failed
An exception occurred within a child process:
  BuildError: Failed executing: /usr/local/Cellar/zig/HEAD-2f538f3_1/bin/zig cc hello.c -o hello
/usr/local/Homebrew/Library/Homebrew/formula.rb:2175:in `block in system'
/usr/local/Homebrew/Library/Homebrew/formula.rb:2111:in `open'
/usr/local/Homebrew/Library/Homebrew/formula.rb:2111:in `system'
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/zig.rb:49:in `block in <class:Zig>'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1975:in `block (3 levels) in run_test'
/usr/local/Homebrew/Library/Homebrew/utils.rb:558:in `with_env'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1974:in `block (2 levels) in run_test'
/usr/local/Homebrew/Library/Homebrew/formula.rb:924:in `with_logging'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1973:in `block in run_test'
/usr/local/Homebrew/Library/Homebrew/mktemp.rb:63:in `block in run'
/usr/local/Homebrew/Library/Homebrew/mktemp.rb:63:in `chdir'
/usr/local/Homebrew/Library/Homebrew/mktemp.rb:63:in `run'
/usr/local/Homebrew/Library/Homebrew/formula.rb:2224:in `mktemp'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1967:in `run_test'
/usr/local/Homebrew/Library/Homebrew/test.rb:43:in `block in <main>'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:93:in `block in timeout'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:33:in `block in catch'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:33:in `catch'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:33:in `catch'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:108:in `timeout'
/usr/local/Homebrew/Library/Homebrew/test.rb:48:in `<main>'
```

### What did you expect to happen?

The tests should pass for `brew install zig --HEAD`.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
> brew install zig --HEAD
> brew test zig
```

### PR DESCRIPTION
**Pull request by kubkon**
This short commit updates the Zig formula to always prefer static
linking of LLVM via `ZIG_STATIC_LLVM` CMake flag.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Closes #78025 

Closes https://github.com/ziglang/zig/issues/8860

cc @Bo98

### CODE REVIEW

**Review by carlocab — APPROVED** [review]
Thanks, @kubkon.

**Comment by carlocab** [comment]
Merged this without bottles, as I assume this change has no impact on users on the stable version of `zig`. If it does, we should open another PR bumping `zig`'s `revision`.
